db/migrate/20210218135141_update_postal_code_registration.rb

```diff
@@ -35,7 +40,9 @@ def postal_code_need_update?(postal_code)
   def reformat_postal_code(postal_code)
     postal_code = postal_code.to_s if postal_code.is_a? Integer
 
-    return "0#{postal_code}" if postal_code.length == 4
+    if postal_code.length == 4 && postal_code.respond_to?(:to_s)
+      return "0#{postal_code}"
+    end
```

Comment:
There is no need to overcheck for type, either postal code is an integer, then you can cast it to a string, or it's an integer, and you may cast it to a string.

Rails will run this migration into a transaction, if this fails we may fix it properly.